cwm_worker_tests/upload_download_test.py

```diff
@@ -42,44 +39,61 @@ def upload(endpoint, access_key, secret_key, bucket, num_files, file_size, outpu
         else:
             print(f'Bucket exists! [{bucket}]')
 
-        start_time = datetime.datetime.now()
-
-        # Create test file, remove at the end
-        filepath = create_test_file(file_size)
-
-        print(f'Uploading files...')
-        for i in range(num_files):
-            filename = get_upload_filename()
-            client.fput_object(bucket, filename, filepath)
-            print(f'\rUploaded: {i+1} files', end='', flush=True)
-
-        end_time = datetime.datetime.now()
-        elapsed_seconds = (end_time - start_time).total_seconds()
-        print(f'\nFiles uploaded! [{elapsed_seconds} seconds]')
-
+        test_filepath = create_test_file(file_size)
+
+        timestamp = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.000%fZ')
+        output_filename = f'upload-report-{timestamp}.csv'
+        output_filepath = output_dir + '/' + output_filename
+
+        with open(output_filepath, 'w', buffering=1) as csvfile:
+            csvfile.write(f'file_index,filename,file_upload_elapsed_time_seconds\n')
+            print(f'Uploading files... [CSV report: {output_filepath}]')
+            for i in range(num_files):
+                index = i+1
+                file_upload_start_time = datetime.datetime.now()
+                filename = get_filename(file_size=file_size, index=index)
+                try:
+                    client.fput_object(bucket, filename, test_filepath)
+                except S3Error as e:
+                    print(f'Exception: {e}')
+                    exit(1)
+                except Exception as e:
+                    print(f'Exception: {e}')
+                    exit(1)
+                file_upload_end_time = datetime.datetime.now()
+                file_upload_elapsed_time_seconds = (file_upload_end_time - file_upload_start_time).total_seconds()
+                print(f'\rUploaded: {index} files', end='', flush=True)
+                csvfile.write(f'{index},{filename},{file_upload_elapsed_time_seconds}\n')
+
+        output_filesize = os.path.getsize(output_filepath)
+        print(f'\nCSV report generated! [{output_filepath}] ({output_filesize} bytes)')
+
+        upload_end_time = datetime.datetime.now()
+        upload_elapsed_time_seconds = (upload_end_time - upload_start_time).total_seconds()
+        print(f'Upload finished! [{upload_elapsed_time_seconds} seconds]')
     except S3Error as e:
         print(f'\nMinIO exception: {e}')
     except Exception as e:
         print(f'\nException: {e}')
```

Comment:
you should remote this try catch too, these errors should be raised and cause the script to stop